### DISCUSSION TITLE
Welcome to TerraFX Discussions!

### DISCUSSION THREAD

**Discussion by tannergooding** [terrafx/terrafx#153] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

### Code of Conduct

TerraFX and everyone contributing (this includes issues, pull requests, the
wiki, etc) must abide by the [CODE_OF_CONDUCT](CODE_OF_CONDUCT.md).
Instances of abusive, harassing, or otherwise unacceptable behavior may be
reported by contacting the project team at admin@terrafx.dev.

**Comment by atcarter714** [comment]
I found this project recently and have been browsing a little bit. I looked through bits and pieces of the code, like some of the math and numerics stuff. I must say, that's some good, clean code I saw there! Overall, it looks really interesting but I'm not familiar with the project structure or exactly what's going on here, I just understand that this is creating a more abstract layer over the top of multiple rendering and multimedia APIs (DirectX, Vulkan, etc) so that applications can utilize their power without worrying about the underlying API-specific complexities ... am I understanding things right? It's a cross-API solution for real-time 2D/3D?

I definitely like what I saw in the implementation of vector types and math. Only thing I might have done differently was like in Vector3 there are readonly properties like `public static Vector3 Zero => Create(0.0f, 0.0f, 0.0f);` and `public static Vector3 UnitX => Create(1.0f, 0.0f, 0.0f);` ... and I wasn't sure why you did it that way instead of something like this:

```
public static readonly Vector3 Zero = default;

public static readonly Vector3 UnitX = new( 1f, 0f, 0f );
```

That made me start wondering if maybe there's something I don't know here, haha, I would have assumed that having a readonly value assigned to it somewhere in RAM would be _slightly_ more efficient that calling something to return a new one each time? Am I wrong, and taking the wrong approach to these kinds of scenarios?!
